### PR TITLE
Fixes JdbcExecutor's uses of RawParameterizedSqlStatement

### LINKED ISSUE #4588: RawParameterizedSqlStatement fails to run on JdbcExecutor
### Search first

- [X] I searched and no similar issues were found

### Description

Calling JdbcExecutor::queryForObject(SqlStatement sql, Class<T> requiredType) with a RawParameterizedSqlStatement fails on Oracle databases with an error about an "invalid index".

Rewriting the query into it's equivalent with a RawSqlStatement works as expected.

Believe this is due to parameters being incorrectly set on the PreparedStatement.
* PreparedStatement's documentation hints that the parameterIndex starts at 1; an index of 0 is used for the first parameter.
* Parameters are retrieved with `parameters.get(0)` and `parameters.get(i)`; the first looks wrong to me.

### Steps To Reproduce

* Build a Liquibase extension or an executable project using liquibase-core.
* Call JdbcExecutor::queryForObject(SqlStatement sql, Class<T> requiredType) with a RawParameterizedSqlStatement.
* Catch and view the exception.

### Expected/Desired Behavior

Calling JdbcExecutor::queryForObject(SqlStatement sql, Class<T> requiredType) with a RawParameterizedSqlStatement should return the query's result.

### Liquibase Version

4.23.0

### Database Vendor & Version

Oracle 18c

### Liquibase Integration

CLI

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

macOS 13.4

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

See #4589 for PR.

### PR DESCRIPTION
**Pull request by andrew-simmons**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description
Adds a private method, JdbcExecutor::setParameters, to set parameters from RawParameterizedSqlStatement to PreparedStatement.
Replaces parameters loop in JdbcExecutor::execute with setParameters. 
Replaces parameters loop in JdbcExecutor::query with setParameters.

Fixes #4588

### CODE REVIEW

File: liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java [hunk 1]
```diff
@@ -180,6 +177,13 @@ public void execute(final SqlStatement sql, final List<SqlVisitor> sqlVisitors)
         execute(new ExecuteStatementCallback(sql, sqlVisitors), sqlVisitors);
     }
 
+    private void setParameters(final PreparedStatement pstmt, final RawParameterizedSqlStatement sql) throws SQLException {
+        final List<?> parameters = sql.getParameters();
+        for (int i = 0; i < parameters.size(); i++) {
+            pstmt.setObject(i + 1, parameters.get(i));
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java:183]: whow, PreparedStatement index starts at 1. Good catch!

**Comment by andrew-simmons** [comment]
Any idea why the package artifacts failed?

File: liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java (hunk 1 above)
Comment (andrew-simmons) [reply liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java:183]: Thanks. Any suggestions or feedback to help get this merged?

**Comment by filipelautert** [comment]
Hello @andrew-simmons - our build process is going through some refactoring and it is blocking me to finish the review . Once it is done I'll move the pr ahead. Thanks for your patience!

**Comment by filipelautert** [comment]
Hello @andrew-simmons ! would you be able to fix the conflict so we can get it merged? thanks!

**Review by filipelautert — APPROVED** [review]

**Pull request merged**